### PR TITLE
infra!: drop NodeJS v14 and v16 support

### PR DESCRIPTION
**Pull request by Shinigami92**
Drop support for NodeJS v14 due to it reached EOL 2023-04-30

This is targeted for v9.0 and can be merged now ✅

### CODE REVIEW

File: scripts/bundle.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ console.log('Building dist for node (cjs)...');
 
 // Generate entry-points for cjs compatibility
 const localeDir = 'locale';
-const target = ['ES2019', 'node14.17'];
+const target = ['ES2020', 'node16.13'];
```
Comment (Shinigami92) [inline scripts/bundle.ts]: according to https://kangax.github.io/compat-table/es2016plus/#node16_11 we can set ES2022

File: scripts/bundle.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { allLocales } from '../src';
 
 console.log('Building dist for node (cjs)...');
 
-const target = ['ES2019', 'node14.17'];
+const target = ['ES2022', 'node16.13'];
```
Comment (Shinigami92) [inline scripts/bundle.ts]: This needs to be node18 ...

**Comment by matthewmayer** [comment]
Per #2521 suggest adding the following to `docs/guide/upgrading_v9/2121.md`:
```
### Node 14 and 16 No Longer Supported

Support for Node.js versions 14 and 16 has been discontinued as these versions have reached their [end-of-life](https://github.com/nodejs/Release). Faker.js 9.0 requires a minimum of Node.js version 18.
```

**Comment by Shinigami92** [comment]
> Per #2521 suggest adding the following to `docs/guide/upgrading_v9/2121.md`:
> 
> ```
> ### Node 14 and 16 No Longer Supported
> 
> Support for Node.js versions 14 and 16 has been discontinued as these versions have reached their [end-of-life](https://github.com/nodejs/Release). Faker.js 9.0 requires a minimum of Node.js version 18.
> ```

Added 👍

**Comment by ST-DDT** [comment]
@Shinigami92 Could you please fix the merge conflicts here?

**Review by ST-DDT — APPROVED** [review]

**Review by xDivisionByZerox — APPROVED** [review]

**Review by matthewmayer — APPROVED** [review]

**Pull request merged**